pkg/prompter/prompter.go

```diff
@@ -0,0 +1,134 @@
+type FileWriter interface {
+	io.Writer
+	Fd() uintptr
+}
+
+// FileReader provides a minimal readable interface for stdin.
+type FileReader interface {
+	io.Reader
+	Fd() uintptr
+}
+
+// New instantiates a new Prompter.
+func New(stdin FileReader, stdout FileWriter, stderr FileWriter) *Prompter {
+	return &Prompter{
+		stdin:  stdin,
+		stdout: stdout,
+		stderr: stderr,
+	}
+}
+
+// Select prompts the user to select an option from a list of options.
+func (p *Prompter) Select(prompt, defaultValue string, options []string) (int, error) {
+	var result int
+	q := &survey.Select{
+		Message:  prompt,
+		Options:  options,
+		PageSize: 20,
+		Filter:   latinMatchingFilter,
+	}
+	if defaultValue != "" {
+		for _, o := range options {
+			if o == defaultValue {
+				q.Default = defaultValue
+				break
+			}
+		}
+	}
+	err := p.ask(q, &result)
+	return result, err
+}
+
+// MultiSelect prompts the user to select multiple options from a list of options.
+func (p *Prompter) MultiSelect(prompt string, defaultValues, options []string) ([]int, error) {
+	var result []int
+	q := &survey.MultiSelect{
+		Message:  prompt,
+		Options:  options,
+		PageSize: 20,
+		Filter:   latinMatchingFilter,
+	}
+	if len(defaultValues) > 0 {
+		validatedDefault := []string{}
+		for _, x := range defaultValues {
+			for _, y := range options {
+				if x == y {
+					validatedDefault = append(validatedDefault, x)
+				}
+			}
+		}
+		q.Default = validatedDefault
+	}
+	err := p.ask(q, &result)
+	return result, err
+}
+
+// Input prompts the user to input a single-line string.
+func (p *Prompter) Input(prompt, defaultValue string) (string, error) {
+	var result string
+	err := p.ask(&survey.Input{
+		Message: prompt,
+		Default: defaultValue,
+	}, &result)
+	return result, err
+}
+
+// Password prompts the user to input a single-line string without echoing the input.
+func (p *Prompter) Password(prompt string) (string, error) {
+	var result string
+	err := p.ask(&survey.Password{
+		Message: prompt,
+	}, &result)
+	return result, err
+}
+
+// Confirm prompts the user to confirm a yes/no question.
+func (p *Prompter) Confirm(prompt string, defaultValue bool) (bool, error) {
+	var result bool
+	err := p.ask(&survey.Confirm{
+		Message: prompt,
+		Default: defaultValue,
+	}, &result)
+	return result, err
+}
+
+func (p *Prompter) ask(q survey.Prompt, response interface{}, opts ...survey.AskOpt) error {
+	opts = append(opts, survey.WithStdio(p.stdin, p.stdout, p.stderr))
+	err := survey.AskOne(q, response, opts...)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("could not prompt: %w", err)
+}
+
+// LatinMatchingFilter returns whether the value matches the input filter.
+// The strings are compared normalized in case.
+// The filter's diactritics are kept as-is, but the value's are normalized,
+// so that a missing diactritic in the filter still returns a result.
+func latinMatchingFilter(filter, value string, index int) bool {
```

Comment:
(nit) Not sure if it matters but will match the function case just in case

```suggestion
// latinMatchingFilter returns whether the value matches the input filter.
// The strings are compared normalized in case.
// The filter's diactritics are kept as-is, but the value's are normalized,
// so that a missing diactritic in the filter still returns a result.
func latinMatchingFilter(filter, value string, index int) bool {
```

Reply:
Yeah this is something that I don't like about godoc comments... they encourage/enforce sentences but also the first letter can be uncapitalized, it bothers me.